php/class-wp-app-manifest.php

```diff
@@ -0,0 +1,202 @@
+		<meta name="theme-color" content="<?php echo esc_attr( $this->get_theme_color() ); ?>">
+		<?php
+	}
+
+	/**
+	 * Gets the theme color for the manifest.
+	 *
+	 * Mainly copied from Jetpack_PWA_Helpers::get_theme_color().
+	 * First looks for the header background color in the AMP for WordPress plugin, if it's active.
+	 * This color displays on loading the app.
+	 *
+	 * @return string $theme_color The theme color for the manifest.json file, as a hex value.
+	 */
+	public function get_theme_color() {
+		if ( current_theme_supports( 'custom-background' ) ) {
+			$background_color = get_background_color(); // This returns a hex value without the leading #, or an empty string.
+			if ( $background_color ) {
+				$theme_color = "#{$background_color}";
+			}
+		}
+
+		if ( ! isset( $theme_color ) ) {
+			$theme_color = self::FALLBACK_THEME_COLOR;
+		}
+
+		/**
+		 * Enables overriding the PWA theme color.
+		 *
+		 * @param string $theme_color Hex color value.
+		 */
+		return apply_filters( 'pwa_background_color', $theme_color );
+	}
+
+	/**
+	 * Gets the manifest data for the REST API response.
+	 *
+	 * Mainly copied from Jetpack_PWA_Helpers::render_manifest_json().
+	 */
+	public function get_manifest() {
+		$manifest = array(
+			'name'      => get_bloginfo( 'name' ),
+			'start_url' => get_home_url(),
+			'display'   => 'minimal-ui',
+			'lang'      => get_locale(),
+			'dir'       => is_rtl() ? 'rtl' : 'ltr',
+		);
+
+		/**
+		 * Gets the 'short_name' by limiting the blog name to 12 characters.
+		 * And if this cuts off a word, it omits the word entirely by using a positive look-ahead in the regex.
+		 * For example, the first 12 characters of 'My PWA WordPress Site' are 'My PWA WordP'.
+		 * Because this cuts off the last word, this removes 'WordP' entirely: 'My PWA'.
+		 *
+		 * @link https://stackoverflow.com/questions/12646197/cut-the-string-to-be-80-characters-and-must-keep-the-words-without-cutting-th#answer-12646400
+		 */
+		preg_match( '/^.{0,12}(?= |$)/', get_bloginfo( 'name' ), $short_name_matches );
+		if ( $short_name_matches ) {
+			$manifest['short_name'] = $short_name_matches[0];
+		}
+
+		$theme_color = $this->get_theme_color();
+		if ( $theme_color ) {
+			$manifest['background_color'] = $theme_color;
+			$manifest['theme_color']      = $theme_color;
+		}
+
+		$description = get_bloginfo( 'description' );
+		if ( $description ) {
+			$manifest['description'] = $description;
+		}
+
+		$manifest_icons = $this->get_icons();
+		if ( $manifest_icons ) {
+			$manifest['icons'] = $manifest_icons;
+		}
+
+		/**
+		 * Enables overriding the manifest json.
+		 *
+		 * There are more possible values for this, including 'orientation' and 'scope.'
+		 * See the documentation: https://developers.google.com/web/fundamentals/web-app-manifest/
+		 *
+		 * @param array $manifest The manifest to send in the REST API response.
+		 */
+		return apply_filters( 'pwa_manifest_json', $manifest );
```

Comment:
This filter would probably be better named just `web_app_manifest`. It's not JSON at this point, but rather an array.